qiskit/visualization/pulse_v2/core.py

```diff
@@ -0,0 +1,493 @@
+            # create drawing objects for waveform
+            for gen in PULSE_STYLE['generator.waveform']:
+                for drawing in sum(list(map(gen, chan_event.get_waveforms())), []):
+                    self._replace_drawing(drawing)
+            # create drawing objects for frame change
+            for gen in PULSE_STYLE['generator.frame']:
+                for drawing in sum(list(map(gen, chan_event.get_frame_changes())), []):
+                    self._replace_drawing(drawing)
+            # create channel info
+            chan_info = types.ChannelTuple(chan, 1.0)
+            for gen in PULSE_STYLE['generator.channel']:
+                for drawing in gen(chan_info):
+                    self._replace_drawing(drawing)
+
+        # create snapshot
+        snapshot_sched = program.filter(instruction_types=[pulse.instructions.Snapshot])
+        for t0, inst in snapshot_sched.instructions:
+            inst_data = types.NonPulseTuple(t0, self.dt, inst)
+            for gen in PULSE_STYLE['generator.snapshot']:
+                for drawing in gen(inst_data):
+                    self._replace_drawing(drawing)
+
+        # create barrier
+        snapshot_sched = program.filter(instruction_types=[pulse.instructions.RelativeBarrier])
+        for t0, inst in snapshot_sched.instructions:
+            inst_data = types.NonPulseTuple(t0, self.dt, inst)
+            for gen in PULSE_STYLE['generator.barrier']:
+                for drawing in gen(inst_data):
+                    self._replace_drawing(drawing)
+
+    def set_time_range(self,
+                       t_start: Union[int, float],
+                       t_end: Union[int, float]):
+        """Set time range to draw.
+
+        The update to time range is applied after py:method`update_channel_property` is called.
+
+        Args:
+            t_start: Left boundary of drawing in units of cycle time or real time.
+            t_end: Right boundary of drawing in units of cycle time or real time.
+
+        Raises:
+            VisualizationError: When times are given in float without specifying dt.
+        """
+        # convert into nearest cycle time
+        if isinstance(t_start, float):
+            if self.dt is not None:
+                t_start = int(t_start / self.dt)
+            else:
+                raise VisualizationError('Floating valued start time %f seems to be in '
+                                         'units of sec but dt is not specified.' % t_start)
+        # convert into nearest cycle time
+        if isinstance(t_end, float):
+            if self.dt is not None:
+                t_end = int(t_end / self.dt)
+            else:
+                raise VisualizationError('Floating valued end time %f seems to be in '
+                                         'units of sec but dt is not specified.' % t_end)
+
+        duration = t_end - t_start
+
+        self.bbox_left = t_start - int(duration * PULSE_STYLE['formatter.margin.left'])
```

Comment:
Should absolute positioning handled in output-independent data container? What if we wanted to draw using text, or 3d? This seems to be coupled to the matplotlib model/drawer?

Reply:
Good point. Perhaps this is coupled to coordinate system of matplotlib. This bbox thing is introduced to handle AbstractCoordinates. Because we generates drawing objects before setting user preference, sometime coordinate is unknown, i.e. pulse channel name should be placed at the most left of the plot. In the plotter interface those abstract coordinates are substituted by those values.